### PR TITLE
add emc2101 component

### PR DESCRIPTION
**Pull request by ellull**
## Description:

Adds EMC2101 component documentation.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4491

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by ellull** [comment]
Waiting for the parent PR esphome/esphome#4491 to be merged.

File: components/emc2101.rst [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        internal_temperature:
+          id: internal_temperature_sensor
+          name: "Internal Temperature Sensor"
+        speed:
+          id: speed_sensor
+          name: "Speed Sensor"
+        duty_cycle:
+          id: duty_cycle_sensor
+          name: "Duty Cycle Sensor"
+        update_interval: 5s
```
Comment (jesserockz) [inline components/emc2101.rst]: This main example should only apply to the main component. 
`sensor` and `output` have their own yaml sections below and we dont need to repeat them here.

If you want to have a full example, please make it the last item on the page in its own section.


```suggestion
.. code-block:: yaml

    # Example configuration entry
    emc2101:
      pwm:
        resolution: 8
```

File: components/emc2101.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+
+Configuration variables:
+************************
+
+- **address** (*Optional*, int): Manually specify the I²C address of the sensor. Defaults to ``0x4C``.
+- **inverted** (*Optional*, boolean): Inverts the output so 0 means 100% duty cycle and 1 means 0%.  Defaults to ``false``.
+- **pwn** (*Optional*): Enable PWN output.
```
Comment (jesserockz) [inline components/emc2101.rst]:
```suggestion
- **pwm** (*Optional*): Enable PWM output.
```

File: components/emc2101.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
+EMC2101 Fan Controller and Temperature sensor
+=============================================
+
+.. seo::
+    :description: Instructions for setting up EMC2101 fan controller and temperature sensor in ESPHome.
+    :image: emc2101.jpg
+    :keywords: EMC2101
+
+The ``emc2101`` component allows you to control a cooling fan from ESPHome (`datasheet <https://ww1.microchip.com/downloads/aemDocuments/documents/MSLD/ProductDocuments/DataSheets/EMC2101-Data-Sheet-DS20006703.pdf>`__). It also provides temperature
+sensors and a tachometer. The fan can be controlled either by a PWN signal or by outputing a DC voltage
```
Comment (jesserockz) [inline components/emc2101.rst]:
```suggestion
sensors and a tachometer. The fan can be controlled either by a PWM signal or by outputing a DC voltage
```

File: components/emc2101.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  - **resolution** (*Optional*, int): Determines effective resolution and the frequency of the PWM signal. Defaults to ``23``.
+  - **divider** (*Optional*, int): Value for the frequency divider. Defaults to ``1``.
+
+- **dac** (*Optional*): Enable DAC output.
+
+  - **conversion_rate** (*Optional*, string): The number of digital to analog conversions perfomed per second. One of ``1/16``, ``1/8``,
+    ``1/4``, ``1/2``, ``1``, ``2``, ``4``, ``8``, ``16``, ``32``. Defaults to ``16``.
+
+
+The PWN signal frequency is determined by the following formula:
```
Comment (jesserockz) [inline components/emc2101.rst]:
```suggestion
The PWM signal frequency is determined by the following formula:
```

File: components/emc2101.rst [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        update_interval: 5s
+
+
+Configuration variables:
+************************
+- **internal_temperature** (*Optional*): The internal temperature sensor.
+
+  - **id** (*Optional*, string): Manually specify the ID for code generation. At least one of **id** and **name** must be specified.
+  - **name** (*Optional*, string): The name for the sensor. At least one of **id** and **name** must be specified.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/emc2101.rst]: This is already defined in the linked sensor config for users to read and learn.
```suggestion
- **internal_temperature** (*Optional*): The internal temperature sensor. 
  All options from :ref:`Sensor <config-sensor>`.
```

File: components/emc2101.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+- **duty_cycle** (*Optional*): Duty cycle sensor.
+
+  - **id** (*Optional*, string): Manually specify the ID for code generation. At least one of **id** and **name** must be specified.
+  - **name** (*Optional*, string): The name for the sensor. At least one of **id** and **name** must be specified.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/emc2101.rst]:
```suggestion
- **external_temperature** (*Optional*): The external temperature sensor. More accurate but it needs an additional transistor to be installed. 
  All options from :ref:`Sensor <config-sensor>`.

- **speed** (*Optional*): The fan speed sensor in RPMs. 
  All options from :ref:`Sensor <config-sensor>`.

- **duty_cycle** (*Optional*): Duty cycle sensor. 
  All options from :ref:`Sensor <config-sensor>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — CHANGES_REQUESTED** [review]
YOu also have not added an entry on the root `index.rst` file, so it will not show up in the tables on the homepage.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**